lib/utils.js

```diff
@@ -0,0 +1,56 @@
+function filterPseudoHeaders (headers) {
+  const dest = {}
+  const headersKeys = Object.keys(headers)
+  let header
+  for (let i = 0; i < headersKeys.length; i++) {
+    header = headersKeys[i]
+    if (header.charCodeAt(0) !== 58) { // fast path for indexOf(':') === 0
+      dest[header.toLowerCase()] = headers[header]
+    }
+  }
+  return dest
+}
+
+function copyHeaders (headers, reply) {
+  const headersKeys = Object.keys(headers)
+
+  let header
+
+  for (let i = 0; i < headersKeys.length; i++) {
```

Comment:
this use of let is __very__ slow on Node 8. Can you please restore `var` everywhere?